File: 1823.cpp
#include<iostream>
#include<vector>

using namespace std;

class Solution {
public:
    int findTheWinner(int n, int k) {
        vector<int> v(n);
        for(int i=0; i<n; i++) v[i] = i+1;
        int idx = 0;
        while(v.size() > 1){
            idx = (idx + k-1) % v.size();
            v.erase(v.begin() + idx);
        }
        return v[0];
    }
};

int main(){
    Solution s;
    int n, k;
    cin >> n >> k;
    cout << s.findTheWinner(n, k) << endl;
    return 0;
}

File: 1823.py
class Solution:
    def findTheWinner(self, n: int, k: int) -> int:
        circle = [i for i in range(1, n+1)]
        idx = 0
        while len(circle) > 1:
            idx = (idx + k - 1) % len(circle)
            circle.pop(idx)
        return circle[0]
        
        
if __name__ == '__main__':
    s = Solution()
    n = int(input())
    k = int(input())
    print(s.findTheWinner(n, k))